File: catkin_ws/src/roboticentry/src/snode.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

void receive_data(const std_msgs::String& msg)
{
    ROS_INFO("Message received : %s", msg.data.c_str());
}


int main (int argc, char **argv)
{
	ros::init(argc, argv, "scnode");
	ros::NodeHandle nh;


	ros::Subscriber sub = nh.subscribe("/communication", 1000, receive_data);

    ros::spin();
}

File: catkin_ws/src/roboticentry/scripts/snode.py
import rospy
from std_msgs.msg import String

def receive_the_data(msg):
    rospy.loginfo("Message received: ")
    rospy.loginfo(msg)

if __name__ == '__main__':
    rospy.init_node('Spython_N')

    sub = rospy.Subscriber("/communication", String, receive_the_data) 

    rospy.spin()
